src/qibocal/auto/serialize.py

```diff
@@ -0,0 +1,31 @@
+import json
+
+
+def deserialize(raw: dict):
+    """Deserialization of nested dict."""
+    for key, value in raw.items():
+        if isinstance(value, dict):
+            raw[key] = deserialize(value)
+
+    return {load(key): value for key, value in raw.items()}
+
+
+def serialize(raw: dict):
+    """JSON-friendly serialization for nested dict."""
+    for key, value in raw.items():
+        if isinstance(value, dict):
+            raw[key] = serialize(value)
+
+    return {
+        key if isinstance(key, str) else json.dumps(key): value
+        for key, value in raw.items()
+    }
```

Comment:
```suggestion
    return {
        key if isinstance(key, str) else json.dumps(key):
        value if not isinstance(value, dict) else serialize(value)
        for key, value in raw.items()
    }
```

In this way:
- you treat in the same way `key` and `value`
- you avoid the double iteration
- you avoid modifying the parameter
- you avoid copying the parameter

It seems a bit weird, but there are languages in which this is the only option (more or less for the reasons above).